Glasgow Class 6 - Man Sang Sin - JS1 - Week 3

ManSangSin opened the pull request.

ManSangSin committed:
completed section 1 exercises

- completed exercises A-E

ManSangSin committed:
fixed exercise E

- added brackets to if statement

ManSangSin committed:
completed section 2 mandatory

- completed mandatory exercises 1-3

Joemwa commented on 1-exercises/E-while-loop-with-array/exercise.js:
```diff
@@ -5,19 +5,26 @@
+  for (let i = 0; i < numberOfBirthdays; i++) {
+    if (BIRTHDAYS[i].includes("July")) {
+      return BIRTHDAYS[i];
+      break;
```
I like your use of the for loop.

ali-nasir-ali commented on 2-mandatory/1-weather-report.js:
```diff
@@ -12,52 +12,53 @@
+    );
+  }
+  return cityTempDataArr;
 }
```
Really like your naming the variable very readable.

ali-nasir-ali commented on 2-mandatory/2-financial-times.js:
```diff
@@ -14,7 +22,18 @@ function potentialHeadlines(allArticleTitles) {
+    }
+  }
+  return shortestTitle;
 }
```
amazing thinking finding the solution, it took a long time for me. 
one tip would be to do more frequent git push.

ali-nasir-ali commented on 2-mandatory/3-stocks.js:
```diff
@@ -48,7 +65,19 @@ function getAveragePrices(closingPricesForAllStocks) {
+    priceChange =
+      closingPricesForIndividualCompany[4] -
+      closingPricesForIndividualCompany[0];
+    // add priceChange value to array priceChangeArr. The value is rounded to 2 decimal using math.round and multiplied by 100 then divided by 100
```
wow that very smart way to solve this one, and amazing work with the comments i should start writing more comments in my code as well.

AppolinFotso reviewed (commented):
Well done @ManSangSin! Your codes are clean and readable. 
I like the way you named your variables, it looks amazing.

ManSangSin committed:
practice adding a new test

- added a test to check the length of array from instructions: JS2 Week1 -Practice write a first test
